### DISCUSSION TITLE
How do you import TypeScript from node_modules?

### DISCUSSION THREAD

**Discussion by Industrial** [vercel/next.js#17685] [Help] [answered]
Hi!

I want to use a module that is being transpiled into es6+. This brings some some problems with Next.js so I thought of circumventing the issue by importing the TypeScript source code directly from my own TypeScript code but I am getting this error:

```
error - ./node_modules/react-financial-charts/src/series/BarSeries.tsx 17:0
Module parse failed: The keyword 'interface' is reserved (17:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
```

Is there a way to make this work? Even a hacky temporary solution would make me very happy. This is the only 'decent' financial charting library that supports/uses TypeScript.

**Comment by jamesmosier** [comment, marked as answer]
You would have to augment the webpack config in your `next.config.js` in order to do this and transpile that module with the babel typescript preset. [Custom Webpack Config docs](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config)

I just installed that `react-financial-charts` library though and did not get that typescript issue. They are exporting ES6 as the entry point to their library, not typescript. Therefore you should just need to use [next-transpiled-modules](https://github.com/martpie/next-transpile-modules) to transpile `react-financial-charts` to work with Next and browsers.
